src/new-components/loading/loading.tsx

```diff
@@ -0,0 +1,58 @@
+import React from 'react'
+import { Box } from '../box'
+import styles from './loading.module.css'
+
+type Size = 'small' | 'medium' | 'large'
+
+type LoaderProps = {
+    /**
+     * The non-visual label used for assistive technologies.
+     */
+    label: string
+    /**
+     * The size of the loading spinner.
+     * @default 'small'
+     */
+    size?: Size
+}
+
+const sizeMapping: Record<Size, number> = {
+    small: 24,
+    medium: 36,
+    large: 48,
+}
+
+function Loading({ size = 'small', label }: LoaderProps) {
+    const numericSize = sizeMapping[size]
```

Comment:
question: Do you think it makes any sense to provide a default value? Like what would happen if we pass a wrong value?

Reply:
I was relying on TypeScript here, but good point. We should provide a fallback to the same value as "small" if an invalid value is passed in some code consuming reactist without type checking in place.